app/services/conversion/TransitAccompanyingDocumentConverter.scala

```diff
@@ -75,4 +81,227 @@ object TransitAccompanyingDocumentConverter extends Converter with ConversionHel
         )
     )
 
+  // TODO pass ref data values here along with main model (IE029Data)
+
+  def fromP5ToViewModel(
+    ie029: IE029,
+    ie015: IE015,
+    countries: Seq[Country],
+    additionalInformation: Seq[AdditionalInformation],
+    kindsOfPackages: Seq[KindOfPackage],
+    previousDocuments: Seq[PreviousDocumentTypes],
+    supportDocuments: Seq[SupportingDocumentTypes],
+    transportDocuments: Seq[TransportDocumentTypes],
+    circumstanceIndicators: Seq[CircumstanceIndicator],
+    customsOffices: Seq[CustomsOffice],
+    controlResults: Seq[ControlResult]
+  ): ValidationResult[viewmodels.TransitAccompanyingDocumentP5TransitionPDF] = {
+
+    ie029.data match {
+      case IE029MessageData(
+            transitOperation,
+            authorisation,
+            officeOfDeparture,
+            officeOfDestination,
+            officesOfTransit,
+            officeOfExit,
+            holderOfTransit,
+            representative,
+            controlResult,
+            guarantee,
+            consignment
+          ) =>
+        (
+          convertConsignor(consignment.Consignor.map(_.toP4), countries),
+          convertConsignee(consignment.Consignee.map(_.toP4), countries)
+        ).mapN {
+          (consignor, consignee) =>
+            def intStringToBool(intString: String) = intString match {
+              case "1" => true
+              case _   => false
+            }
+
+            viewmodels.TransitAccompanyingDocumentP5TransitionPDF(
```

Comment:
This is starting to look big, can we try and split it up a bit. Maybe for each parameter in the case class `TransitAccompanyingDocumentP5TransitionPDF` we have single line or else defer to a function which handles the logic